internal/cli/atlas/deployments/setup.go

```diff
@@ -237,38 +276,284 @@ func (opts *SetupOpts) waitConnection(port int) error {
 	return errors.New("waitConnection failed")
 }
 
-func getPortForNewLocalCluster(existingContainers []podman.Container) int {
-	maxPort := startHostPort - 1
-	for _, c := range existingContainers {
-		for _, p := range c.Ports {
-			if maxPort < p.HostPort {
-				maxPort = p.HostPort
+func (opts *SetupOpts) promptSettings() error {
+	p := &survey.Select{
+		Message: "How do you want to setup your local MongoDB database?",
+		Options: settingOptions,
+		Default: opts.settings,
+		Description: func(value string, index int) string {
+			return settingsDescription[value]
+		},
+	}
+
+	return telemetry.TrackAskOne(p, &opts.settings, nil)
+}
+
+func (opts *SetupOpts) generateDeploymentName() {
+	opts.DeploymentName = fmt.Sprintf("local%v", rand.Intn(10000)) //nolint // no need for crypto here
+}
+
+func (opts *SetupOpts) promptDeploymentName() error {
+	p := &survey.Input{
+		Message: "Cluster Name [This can't be changed later]",
+		Default: opts.DeploymentName,
+	}
+
+	return telemetry.TrackAskOne(p, &opts.DeploymentName, survey.WithValidator(func(ans interface{}) error {
+		name, _ := ans.(string)
+		return validateDeploymentName(name)
+	}))
+}
+
+func (opts *SetupOpts) promptMdbVersion() error {
+	p := &survey.Select{
+		Message: "MongoDB Version",
+		Options: mdbVersions,
+		Default: opts.MdbVersion,
+	}
+
+	return telemetry.TrackAskOne(p, &opts.MdbVersion, nil)
+}
+
+func validatePort(p int) error {
+	if p <= 0 || p > 65535 {
+		return errors.New("port must within the range 1..65535")
+	}
+	return nil
```

Comment:
[not blocking] Could we also check if the port is in use here as well? It would be great to reject already used ports, either from processes or other deplotments that have been configured, before the setup starts & then fails, requiring the user to start again.

Reply:
captured in CLOUDP-197638

Reply:
Thank you!